chain/pool/txpool.go

```diff
@@ -46,12 +53,19 @@ type TxnPool struct {
 
 	sync.RWMutex
 	lastDroppedTxn *transaction.Transaction
+
+	// sync txn
+	latSyncTime          int64    // last sync time in second
+	mapTxnWithTime       sync.Map // FIFO txn of last MaxSyncTxnInterval seconds
+	txnWithTimeCount     int      // simple counter of mapTxnWithTie
```

Comment:
You need to handle concurrency for ALL added logic

Reply:
yes.